rllib/env/env_runner.py

```diff
@@ -0,0 +1,174 @@
+    Use the `as_remote()` method to create the corresponding Ray remote class. Then
+    instantiate n Actors using the Ray `[ctor].remote(...)` syntax.
+    - EnvRunner clients can get information about the server/node on which the
+    individual Actors are running.
+    """
+
+    @classmethod
+    def as_remote(
+        cls,
+        num_cpus: Optional[int] = None,
+        num_gpus: Optional[Union[int, float]] = None,
+        memory: Optional[int] = None,
+        resources: Optional[dict] = None,
+        max_num_worker_restarts: int = 0,
+    ) -> type:
+        """Returns the EnvRunner class as a `@ray.remote` using given options.
+
+        The returned class can then be used to instantiate ray actors.
+
+        Examples:
+            >>> remote_cls = EnvRunner.as_remote(num_cpus=1, num_gpus=0)
+            >>> array = [remote_cls.remote({"key": "value"}) for _ in range(10)]
+            >>> results = [env_runner.sample.remote() for env_runner in array]
+
+        Args:
+            num_cpus: The number of CPUs to allocate for the remote actor.
+            num_gpus: The number of GPUs to allocate for the remote actor.
+                This could be a fraction as well.
+            memory: The heap memory request in bytes for this task/actor,
+                rounded down to the nearest integer.
+            resources: The default custom resources to allocate for the remote
+                actor.
+            max_num_worker_restarts: The maximum number of restarts a failed EnvRunner
+                will undergo. Note that this setting is only effective if the EnvRunner
+                is a) a Ray Actor and b) managed by a fault-tolerant
+                ray.rllib.utils.actor_manager::FaultTolerantActorManager.
+
+        Returns:
+            The `@ray.remote` decorated EnvRunner class.
+        """
+        return ray.remote(
+            num_cpus=num_cpus,
+            num_gpus=num_gpus,
+            memory=memory,
+            resources=resources,
+            # Automatically restart failed EnvRunners.
+            max_restarts=max_num_worker_restarts,
+        )(cls)
+
+    def __init__(self, *, config, **kwargs):
+        """Initializes an EnvRunner instance.
+
+        Args:
+            config: The config to use to setup this EnvRunner.
+            **kwargs: Forward compatibility kwargs.
+        """
+        self.config = config
+        super().__init__(**kwargs)
+
+    @abc.abstractmethod
+    def assert_healthy(self):
+        """Checks that self.__init__() has been completed properly.
+
+        Useful in case an `EnvRunner` is run as @ray.remote (Actor) and the owner
+        would like to make sure the Ray Actor has been properly initialized.
+
+        Raises:
+            AssertionError: If the EnvRunner Actor has NOT been properly initialized.
+        """
+
+    @abc.abstractmethod
+    def sample(
+        self,
+        *,
+        num_timesteps: Optional[int] = None,
```

Comment:
Is this API needed? I noticed that RW completely override the signature.

Reply:
True, I'm not sure either about the integration of RW into this. Maybe we should retro-fix its sample method into this API (it wouldn't break anything). Alternatively, we could just give the RW version `**kwargs` and then warn if a user sets any of these kwargs.

I do feel that these new arg options of the base `EnvRunner` class are important.
* They allow flexibility by supporting overrides to the default behavior imposed by `self.config`.
* Allows for quick testing/evaluation runs of n timesteps (or n episodes), with exploration or w/o.
* In short, it allows me to use my once configured EnvRunner for different scenarios w/o needing a new EnvRunner with a different config. Like right now, I feel like we cannot really use the main training RolloutWorkers for a quick eval check, you need to setup an extra eval worker set for this, which is tedious. Maybe I just want to pause my training iterations and perform a quick n episode test run (no exploration) to see, where I'm at right now with learning progress.

Reply:
This has been done. I made the API even more minimal now. Just `sample(self, **kwargs)`. The rest can be implemented as needed by subclasses.